// @syncfusion/ej2-react-filemanager/dist/es6/ej2-react-filemanager.es2015.js
import { ComplexBase, applyMixins, ComponentBase } from '@syncfusion/ej2-react-base';
export { Inject } from '@syncfusion/ej2-react-base';
import { createElement, Component } from 'react';
import { FileManager } from '@syncfusion/ej2-filemanager';
export * from '@syncfusion/ej2-filemanager';

class ToolbarItemDirective extends ComplexBase {
}
ToolbarItemDirective.moduleName = 'toolbarItem';
class ToolbarItemsDirective extends ComplexBase {
}
ToolbarItemsDirective.propertyName = 'toolbarItems';
ToolbarItemsDirective.moduleName = 'toolbarItems';

/**
 Represents the Essential JS 2 react FileManager Component.
 * ```tsx
 * <FileManagerComponent showThumbnail={false}></FileManagerComponent>
 * ```
 */
class FileManagerComponent extends FileManager {
    constructor(props) {
        super(props);
        this.initRenderCalled = false;
        this.checkInjectedModules = true;
        this.directivekeys = { 'toolbarItems': 'toolbarItem' };
        this.statelessTemplateProps = null;
        this.templateProps = null;
        this.immediateRender = false;
        this.portals = [];
    }
    render() {
        if (((this.element && !this.initRenderCalled) || this.refreshing) && !this.isReactForeceUpdate) {
            super.render();
            this.initRenderCalled = true;
        }
        else {
            return createElement('div', this.getDefaultAttributes(), [].concat(this.props.children, this.portals));
        }
    }
}
applyMixins(FileManagerComponent, [ComponentBase, Component]);

export { FileManagerComponent, ToolbarItemDirective, ToolbarItemsDirective };
